neighbour-service/src/main/java/no/vegvesen/ixn/federation/service/NeighbourService.java

```diff
@@ -452,6 +462,26 @@ public void createListenerEndpoint(Subscription subscription, Neighbour neighbou
 		return neighbourRepository.findNeighboursByOurRequestedSubscriptions_Subscription_SubscriptionStatusIn(SubscriptionStatus.CREATED);
 	}
 
+	public void updateLocalSubscriptions(Self self) {
+		Set<LocalSubscription> selfSubscriptions = self.getLocalSubscriptions();
+		List<Neighbour> neighboursWithOurRequestedSubscriptions = listNeighboursToConsumeMessagesFrom();
+
+		for(Neighbour neighbour : neighboursWithOurRequestedSubscriptions) {
+			Set<Subscription> subscriptions = neighbour.getOurRequestedSubscriptions().getAcceptedSubscriptionsWithCreateNewQueue();
+			for(Subscription subscription : subscriptions){
+				if(subscription.getBrokerUrl() != null){
+					LocalSubscription localSubscription = selfSubscriptions.stream()
```

Comment:
Will this work with several Subscriptions with the same selector? Why not filter on the queueConsumerUser first?

Reply:
I removed this method. It is not used because we decided to create a new Service for ServiceProvider and we are handling the same functionality there.